shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -568,22 +563,27 @@ object Types {
     def selectIn(name: SignedName, in: TypeRef): TermRef =
       TermRef(this, LookupIn(in, name))
 
-    // overridden in package references
-    override def resolveToSymbol(using Context): Symbol = designator match {
-      case sym: Symbol => sym
-      case LookupIn(pre, name) =>
-        val sym = TermRef(pre, name).resolveToSymbol
-        designator = sym
+    final def symbol(using Context): Symbol =
+      val local = mySymbol
+      if local != null then local
+      else
+        val computed = computeSymbol()
+        mySymbol = computed
+        designator = computed
+        computed
+
+    private def computeSymbol()(using Context): Symbol = designator match
+      case sym: Symbol =>
         sym
+      case LookupIn(pre, name) =>
+        TermRef(pre, name).symbol
       case Scala2ExternalSymRef(owner, path) =>
-        val sym = path.foldLeft(owner) { (owner, name) =>
-          val cls = if owner.isTerm then owner.declaredType.asInstanceOf[Symbolic].symbol else owner
+        path.foldLeft(owner) { (owner, name) =>
+          val cls = if owner.isTerm then owner.declaredType.asInstanceOf[TypeRef].symbol else owner
```

Comment:
maybe a comment to say that owner.isTerm can only be module value so its declared type will always be the TypeRef to its module class

Reply:
Yes, good point. I added a comment.